### PR TITLE
glog.rb add test, use cmake build to generate cmake config files.

### PR DESCRIPTION
**Pull request by ahundt**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Upstream issues:
https://github.com/facebook/folly/issues/648
https://github.com/google/glog/issues/196

### CODE REVIEW

**Comment by ahundt** [comment]
Hmm, should this be solved by adding an extra symlink or some other way? 

How would I add the symlink to resolve the linker errors like what follows?

```
Error Message

failed: brew linkage --test folly
Stacktrace

        Missing libraries:
  /usr/local/opt/glog/lib/libglog.0.dylib
```

File: Formula/glog.rb [hunk 1]
```diff
@@ -12,11 +13,29 @@ class Glog < Formula
...
 
+  depends_on "cmake" => :build
   depends_on "gflags"
 
   def install
-    system "./configure", "--disable-dependency-tracking",
-                          "--prefix=#{prefix}"
-    system "make", "install"
+    mkdir "glog-build" do
+      args = std_cmake_args
```
Comment (dunn) [inline Formula/glog.rb]: If you're not adding any extra arguments, this assignment isn't necessary.

**Comment by dunn** [comment]
> Hmm, should this be solved by adding an extra symlink or some other way?

That's probably something that should be reported upstream.

**Comment by ahundt** [comment]
> That's probably something that should be reported upstream.

Done. But I assume there is some one liner that can create it manually in brew? That'd let me move on. :-)

**Comment by dunn** [comment]
Yeah, you should be able to do something like `lib.install_symlink "libglog.dylib" => "libglog.0.dylib"`

File: Formula/glog.rb [hunk 2]
```diff
@@ -12,11 +13,29 @@ class Glog < Formula
...
 
   def install
-    system "./configure", "--disable-dependency-tracking",
-                          "--prefix=#{prefix}"
-    system "make", "install"
+    mkdir "glog-build" do
+      system "cmake", "..", *std_cmake_args
+      system "make", "install"
+    end
+    lib.install_symlink "libglog.dylib" => "libglog.0.dylib"
```
Comment (dunn) [inline Formula/glog.rb]: My bad, this needs to be `lib/"libglog.dylib" => "libglog.0.dylib"`.  Could you also add a comment with a link to the upstream issue, so we know when to remove this line?

**Comment by dunn** [comment]
Sorry, I think the symlink approach might be totally wrong.

If the library version changed, then instead what we need to do is increment the `revision` on the dependent formulae so they're rebuilt against the new version.

Apologies again for not thinking clearly and sending you in the wrong direction.

**Comment by ahundt** [comment]
no worries! I'm a little confused by your last two comments, what change should I actually make now?

**Comment by dunn** [comment]
For the formulae with errors here, increment their `revision` line or add a `revision 1`: https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/2687/#showFailuresLink

**Comment by ahundt** [comment]
not actually stale... I'll address this when I'm able to

**Comment by ahundt** [comment]
still not stale...

**Comment by ilovezfs** [comment]
I think we should hold off switching this over to CMake until the library situation is sorted out upstream. We made a similar switch to CMake for snappy and had problems for that very reason.

CC @pwnall maybe you can nudge the glog team since this seems to be the same issue that snappy had?

**Comment by ilovezfs** [comment]
It'll need revision bumps on ceres-solver, folly, gjstest, and osquery.

**Comment by ahundt** [comment]
thanks, for some reason the jenkins ci results were empty and I couldn't see what had failed anymore. Hmm, that's a lot of updates... Just revision bumps or actual upstream code changes?

**Comment by ilovezfs** [comment]
>thanks, for some reason the jenkins ci results were empty and I couldn't see what had failed anymore. Hmm, that's a lot of updates... Just revision bumps or actual upstream code changes?

Probably just revision bumps, but we won't actually know until you, CI, or both attempt to build them :)

**Comment by ilovezfs** [comment]
@ahundt what's the status here?

**Comment by ahundt** [comment]
Well it seems the problem is some libraries that depend on this don't look for the correct file name. We could simply add the symlink workaround mentioned earlier so this isn't held up by downstream stuff... is there an easy command to add the symlink?

**Comment by ahundt** [comment]
@ilovezfs is it sensible to create a symlink to work around the problem, and if so, how?

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@ahundt not a symlink, no. We either need to get upstream to fix it so that changing to cmake doesn't change the library version and compatibility numbers, or we need to revision bump each of the reverse dependencies.

**Comment by ahundt** [comment]
ok, I did report upstream in both places but there seems to not be any progress:
https://github.com/facebook/folly/issues/648
https://github.com/google/glog/issues/196

**Comment by ilovezfs** [comment]
@ahundt ok so it sounds like revision bumping the reverse dependencies is what we'll need to do then if we want to use cmake. Here's how to do that: https://github.com/Homebrew/homebrew-core/pull/17807/files

Let me know if you need more guidance.

**Comment by ilovezfs** [comment]
@ahundt feel free to re-open when it includes the revision bumps. Thanks for the PR.

**Comment by ahundt** [comment]
How do I do that? Is there a brew command?

**Comment by ilovezfs** [comment]
You manually edit each formula that depends on this and either add `revision 1` to it (we start counting from zero implicitly) or increment the existing revision by one, so `revision 2` becomes `revision 3`. One commit per formula. Commit subject `foo: revision for bar`

**Comment by ahundt** [comment]
@ilovezfs ok I tried performing the bumps

**Comment by ahundt** [comment]
oh shoot one commit per formula... whoops

**Comment by ahundt** [comment]
@ilovezfs okay I did separate revision bumps now

**Comment by ahundt** [comment]
I think everything is bumped correctly now and the cmake settings of ceres-solver.rb have been updated accordingly.

**Pull request closed without merging**